_plugins/search.rb

```diff
@@ -11,14 +11,27 @@ def self.build_index(site)
         'assets', 'js', 'vendor', 'lunr.js', 'lunr.js'))
       cxt.load(File.join(site.source, '_plugins', 'search.js'))
       cxt.eval("var corpus = #{corpus_page.content};")
-      cxt.eval('corpus.entries.forEach(function(page) {index.add(page);});')
+      cxt.eval("var url_to_doc = {};")
+      cxt.eval(
+        'corpus.entries.forEach(function(page) {' +
+        '  index.add(page);' +
+        '  url_to_doc[page.url] = {url: page.url, title: page.title};' +
+        '});')
```

Comment:
I'd move these lines into `_plugins/search.js`, and load that file after the corpus is passes. Less JS in Ruby files!

```ruby
cxt = V8::Context.new
cxt.load(File.join(site.source, 'assets', 'js', 'vendor', 'lunr.js', 'lunr.js'))
cxt.eval("var corpus = #{corpus_page.content};")
cxt.load(File.join(site.source, '_plugins', 'search.js'))
```